src/tests/Loader/classloader/explicitlayout/Regressions/100220/Runtime_100220.cs

```diff
@@ -0,0 +1,189 @@
+            Assert.Equal(32, GetManagedSize(() => new SubclassOfBaseWithGcSeq()));
+
+            // Structs
+            Assert.Equal(16, Unsafe.SizeOf<MyStructAuto>());
+            Assert.Equal(16, Unsafe.SizeOf<MyStruct9Seq>());
+            Assert.Equal(16, Unsafe.SizeOf<MyStruct9Exp>());
+            Assert.Equal(16, Unsafe.SizeOf<MyStruct1000Seq>());
+            Assert.Equal(1000, Unsafe.SizeOf<MyStruct1000Exp>());
+        }
+        else
+        {
+            // Classes
+            Assert.Equal(16, GetManagedSize(() => new MyClassAuto()));
+            Assert.Equal(20, GetManagedSize(() => new MyClass1000Exp()));
+            Assert.Equal(16, GetManagedSize(() => new MyClass1000Seq()));
+            Assert.Equal(20, GetManagedSize(() => new MyClass1000NoGcExp()));
+            Assert.Equal(1008, GetManagedSize(() => new MyClass1000NoGcSeq()));
+            Assert.Equal(28, GetManagedSize(() => new BaseClassSeq()));
+            Assert.Equal(44, GetManagedSize(() => new SubclassSeq()));
+            Assert.Equal(52, GetManagedSize(() => new SubclassSubclassSeq()));
+            Assert.Equal(36, GetManagedSize(() => new SubclassWithGcSeq()));
+            Assert.Equal(16, GetManagedSize(() => new SubclassOfBaseWithGcSeq()));
+
+            // Structs
+            Assert.Equal(8, Unsafe.SizeOf<MyStructAuto>());
+            Assert.Equal(8, Unsafe.SizeOf<MyStruct9Seq>());
+            Assert.Equal(12, Unsafe.SizeOf<MyStruct9Exp>());
+            Assert.Equal(8, Unsafe.SizeOf<MyStruct1000Seq>());
+            Assert.Equal(1000, Unsafe.SizeOf<MyStruct1000Exp>());
+        }
+
+        // Field offsets:
+        Assert.Equal("(5, 3, 4, 1, 2, 6)", FieldOffsets(new SubclassSubclassSeq { A = 1, B = 2, C = 3, D = 4, E = 5, F = 6 }).ToString());
+        return 100;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static object FieldOffsets(SubclassSubclassSeq f) => (f.E, f.C, f.D, f.A, f.B, f.F);
+
+    [StructLayout(LayoutKind.Sequential, Size = 20)]
+    public class BaseClassSeq
+    {
+        public byte A;
+        public nint B;
+    }
+
+    [StructLayout(LayoutKind.Sequential, Size = 20)]
+    public class BaseClassWithGcSeq
+    {
+        public byte A;
+        public string B;
+    }
+
+    public class SubclassOfBaseWithGcSeq : BaseClassWithGcSeq
+    {
+        public byte C;
+    }
+
+    [StructLayout(LayoutKind.Sequential, Size = 16)]
+    public class SubclassSeq : BaseClassSeq
+    {
+        public byte C;
+        public nint D;
+    }
+
+    [StructLayout(LayoutKind.Sequential, Size = 32)]
+    public class SubclassWithGcSeq : BaseClassSeq
+    {
+        public byte C;
+        public object D;
+    }
+
+    public class SubclassSubclassSeq : SubclassSeq
+    {
+        public byte E;
+        public nint F;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.AggressiveOptimization)]
+    private static int GetManagedSize(Func<object> allocator)
```

Comment:
This is not reliable as you have pointed out. 

I think a reliable version of this check would be to inherit from the type you want to check, add a byte field, and verify offset of the byte field.

Reply:
Would an `UnsafeAccessor` to `RuntimeHelpers.GetRawObjectDataSize` work better here instead maybe?